cmd/goplugin-terraform/resource/configuration.go

```diff
@@ -0,0 +1,150 @@
+package resource
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/lyraproj/servicesdk/serviceapi"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-hclog"
+	"github.com/lyraproj/pcore/px"
+	"github.com/lyraproj/pcore/serialization"
+	"github.com/lyraproj/pcore/types"
+)
+
+// Config is a terraform configuration identified by a working directory
+type Config struct {
+	WorkingDir string
+	UniqueID   *string //HACK to force Read to return a new value, in order to always trigger an Update
+	Output     *px.Value
+}
+
+// ConfigHandler is used to perform CRUD operations on a Config resource
+type ConfigHandler struct{}
+
+// Create a new set of resources
+func (*ConfigHandler) Create(desiredState *Config) (*Config, string, error) {
+	log := hclog.Default()
+	log.Debug("Creating Config", "desiredState", desiredState)
+	externalID := extID(*desiredState)
+	if ok, err := dirExists(externalID); !ok {
+		log.Debug("error reading directory", "err", err)
+		return nil, "", serviceapi.NotFound("LoadBalancer", externalID)
+	}
+	return desiredState, externalID, apply(desiredState)
+}
+
+// Read an existing Config
+func (*ConfigHandler) Read(externalID string) (*Config, error) {
+	log := hclog.Default()
+	log.Debug("Reading Config", "externalID", externalID)
+	if ok, err := dirExists(externalID); !ok {
+		log.Debug("error reading directory", "err", err)
+		return nil, serviceapi.NotFound("LoadBalancer", externalID)
+	}
+	//HACK return a new uniqueID to ensure update is always called
+	s := uniqueString()
+	return &Config{
```

Comment:
The `Config.WorkingDir` should be set equal to the `externalID` here I guess?

Reply:
👍